src/epgsql.erl

```diff
@@ -259,8 +265,11 @@ parse(C, Sql) ->
 parse(C, Sql, Types) ->
     parse(C, "", Sql, Types).
 
--spec parse(connection(), iolist(), sql_query(), [epgsql_type()]) ->
-                   {ok, #statement{}} | {error, query_error()}.
+-spec parse(connection(), iolist(), sql_query(), [epgsql_type()]) -> Reply
+  when
+    Reply :: {ok, #statement{}}
+           | {error, query_error()}
+           | {error, socket_error()}.
```

Comment:
Maybe `{error, query_error() | socket_error()}`? I like both ways, but maybe you have some opinion on that?